File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.py
# Last updated: 1/8/2026, 5:29:42 p.m.
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        h = {}
        m = 0
        cont = 0
        inicio = 0

        for i in range(len(s)):
            if s[i] in h:
                if m < cont: m = cont
                cont = i - h[s[i]] - 1
                for j in range(inicio, h[s[i]]):
                    del h[s[j]]
                    inicio += 1
                inicio += 1
                

            cont += 1
            h[s[i]] = i
        
        return m if m > cont else cont

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.c
// Last updated: 1/8/2026, 5:29:45 p.m.
int lengthOfLongestSubstring(char* s) {
    int abc[128]; 
    int i = 0, start = -1, maxLength = 0;
    char *aux; 
    memset(abc, -1, sizeof(abc));

    while(s[i] != '\0'){
        if(abc[s[i]] > start)
            start = abc[s[i]];

        abc[s[i]] = i;
        if(i-start > maxLength) maxLength = i-start;
        i++;
    }

    return maxLength;
}